friendlist/src/main/java/com/bytro/friendlist/handler/FriendRequestHandler.java

```diff
@@ -0,0 +1,101 @@
+
+    private final FriendRequestService friendRequestService;
+    private final FriendRequestMapper friendRequestMapper;
+
+    private final EmailService emailService;
+
+    private final MessageSource messageSource;
+
+    @Value("${request.mail.subject}")
+    private String mailSubject;
+
+    @Value("${request.mail.receiver}")
+    private String mailReceiver;
+
+    FriendRequestHandler(
+            final FriendRequestService friendRequestService,
+            final FriendRequestMapper friendRequestMapper,
+            final MessageSource messageSource,
+            final EmailService emailService) {
+        this.friendRequestService = friendRequestService;
+        this.friendRequestMapper = friendRequestMapper;
+        this.messageSource = messageSource;
+        this.emailService = emailService;
+    }
+
+    public BaseResponse<FriendRequestResponse> send(SendFriendRequest sendFriendRequest) {
+        validateFriendRequest(sendFriendRequest);
+        FriendRequest friendRequest = friendRequestMapper.requestToEntity(sendFriendRequest);
+        FriendRequest requestSent = friendRequestService.send(friendRequest);
+        FriendRequestResponse friendRequestResponse =
+                friendRequestMapper.entityToResponse(requestSent);
+        String message =
+                "Hi user, Configurations!! you will get a new friend..\n"
+                        + "Here is a special message for you : \n"
+                        + sendFriendRequest.message();
+        emailService.sendSimpleMail(new EmailDetails(mailReceiver, message, mailSubject));
+        return new BaseResponse<>(
+                ResultCode.SUCCESS.getValue(),
+                messageSource.getMessage(
+                        "friend.request.sent.successfully", new String[] {}, Locale.US),
+                friendRequestResponse);
+    }
+
+    private void validateFriendRequest(SendFriendRequest sendFriendRequest) {
+        if (sendFriendRequest.senderId() == sendFriendRequest.receiverId()) {
+            throw new CustomException(
+                    ResultCode.SENDER_RECEIVER_SAME.getValue(),
+                    messageSource.getMessage(
+                            "send.receiver.should.not.be.same", new String[] {}, Locale.US));
+        }
+        checkUser(sendFriendRequest.senderId());
+        checkUser(sendFriendRequest.receiverId());
+    }
+
+    private void checkUser(int userId) {
+        if (!UserUtils.checkUsersExistence(userId)) {
+            throw new CustomException(
+                    ResultCode.USER_NOT_FOUND.getValue(),
+                    messageSource.getMessage("user.not.found", new String[] {}, Locale.US));
+        }
+    }
+
+    public BaseResponse<Void> acceptRejectFriendRequest(
+            AcceptRejectFriendRequest acceptRejectFriendRequest) {
+        FriendRequest friendRequest =
+                friendRequestMapper.requestToEntity(acceptRejectFriendRequest);
+        if (acceptRejectFriendRequest.status()) {
```

Comment:
why we need two method to accept and reject?

Reply:
in case of business logic changes it becomes to  messy so make it separate will help us in readability